engine/storage/src/main/java/org/apache/cloudstack/storage/allocator/AbstractStoragePoolAllocator.java

```diff
@@ -207,6 +215,13 @@ protected boolean filter(ExcludeList avoid, StoragePool pool, DiskProfile dskCh,
             return false;
         }
 
+        if (diskProvisioningTypeStrictness){
+            StoragePoolDetailVO hardwareAcceleration = storagePoolDetailsDao.findDetail(pool.getId(), "hardwareAccelerationSupported");
+            if (!dskCh.getProvisioningType().equals("thin") && hardwareAcceleration == null || !hardwareAcceleration.getValue().equals("true")){
```

Comment:
can you move this to a generic method, something like _supportsDiskProvisioingType_ , and better check for the hypervisor (as this is only for VMware) and other required checks.

Reply:
Done